test/load/locust-files/gameserver_allocation.py

```diff
@@ -0,0 +1,160 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from locust import HttpLocust, TaskSet, clients, events, task
+import locust.events
+import json
+import time
+import socket
+import atexit
+
+FLEET_SIZE = 100
+FLEET_NAME = "scale-test-fleet"
+FLEET_RESOURCE_PATH = "/apis/stable.agones.dev/v1alpha1/namespaces/default/fleets"
+ALLOCATION_RESOURCE_PATH = "/apis/stable.agones.dev/v1alpha1/namespaces/default/gameserverallocations"
+
+class UserBehavior(TaskSet):
+    @task
+    def allocate_game_server(self):
+        # Get the number of ready game servers.
+        fleet_self_link = FLEET_RESOURCE_PATH + "/" + FLEET_NAME
+        response = self.client.get(str(fleet_self_link))
+        response_json = response.json()
+        ready_replicas = response_json['status']['readyReplicas']
+        # Allocate game servers.
+        payload = {
+            "apiVersion":"stable.agones.dev/v1alpha1",
+            "kind":"GameServerAllocation",
+            "metadata":{
+                "generateName":"gs-allocation-",
+                "namespace":"default"
+            },
+            "spec":{
+                "required":{
+                    "matchLabels":{
+                        "stable.agones.dev/fleet":FLEET_NAME
+                    }
+                }
+            }
+        }
+        headers = {'content-type': 'application/json'}
+        start_time = time.time()
+        response = self.client.post(str(ALLOCATION_RESOURCE_PATH), data=json.dumps(payload), headers=headers)
+        events.request_success.fire(request_type="ReadyReplicas", name="ReadyReplicas", response_time=ready_replicas, response_length=0)
+        # Wait for allocation.
+        while True:
+            response_json = response.json()
+            status = response_json.get('status')
+            if (status != None):
+                game_server_state = response_json['status']['state']
+                if (game_server_state == "Allocated"):
+                    total_time = int((time.time() - start_time) * 1000)
+                    events.request_success.fire(request_type="GameServerAllocated", name="GameServerAllocated", response_time=total_time, response_length=0)
+                    break
+                elif (game_server_state == "UnAllocated"):
+                    total_time = int((time.time() - start_time) * 1000)
+                    events.request_success.fire(request_type="GameServerUnAllocated", name="GameServerUnAllocated", response_time=total_time, response_length=0)
```

Comment:
Same as in other file. This file violates pep8 stadard. Line is too long, should be split into multiple.

Reply:
Done. Thanks.